pkg/server/storage/mysql.go

```diff
@@ -55,6 +55,42 @@ func (my *MysqlStorage) QueryHostkeys() (ResultIterator, error) {
 	return &SqlResultIterator{Rows: rows}, nil
 }
 
+func (my *MysqlStorage) RecordGitHubMapping(mapping map[string]string) error {
+	// Prepare for batch insert
+	entries := make([]string, 0, len(mapping))
+	values := make([]interface{}, 0, len(mapping)*2)
+	for ssoIdentity, githubUser := range mapping {
+		// Create one set of values for each mapping
+		entries = append(entries, "(?, ?)")
+		// Append matching values for mapping
+		values = append(values, ssoIdentity)
+		values = append(values, githubUser)
+	}
+	// Create query with necessary number of (?, ?) values
+	// Note: The use of VALUES in VALUES(github_username) will be deprecated in MYSQL 8.0.20
+	// https://dev.mysql.com/doc/refman/8.0/en/insert-on-duplicate.html
+	stmt := fmt.Sprintf(
+		"INSERT INTO github_user_mappings (sso_identity, github_username) VALUES %s ON DUPLICATE KEY UPDATE github_username=VALUES(github_username)",
```

Comment:
Seems that it handles only new mapping or updated mappings, but not deleted - specifically when user leave the organization that user won't be returned in the call to GraphQL API and will remain in database forever. Can't think of any downside to it, but probably worth mentioning explicitly.

Reply:
ahh yeah I hadn't thought of that